app/code/Magento/Downloadable/Observer/UpdateLinkPurchasedObserver.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Downloadable\Observer;
+
+use Magento\Framework\Event\ObserverInterface;
+
+/**
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class UpdateLinkPurchasedObserver implements ObserverInterface
```

Comment:
Hi @Jitheesh. Thank you for your contribution. Could you explain, please, is there any reason why the observer was created instead of fixing the issue directly in the code where it appears? 
Thank you

Reply:
Hi @rogyar 

This issue is related to downloadable component and issue is appears at sales component, 
https://github.com/magento/magento2/blob/ab66a14ae164941b1b54ee6ed5071076816d29cb/app/code/Magento/Sales/Model/Order/CustomerManagement.php#L99-L125

so I thought it would be good to add this fix on corresponding component.  We already have other Observer based on this rule.
https://github.com/magento/magento2/blob/ab66a14ae164941b1b54ee6ed5071076816d29cb/app/code/Magento/Downloadable/Observer/SetLinkStatusObserver.php#L11